django/thunderstore/repository/tasks.py

```diff
@@ -1,12 +1,48 @@
 from celery import shared_task
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from ulid2 import generate_ulid_as_uuid
 
 from thunderstore.repository.api.experimental.tasks import (
     update_api_experimental_caches,
 )
 from thunderstore.repository.api.v1.tasks import update_api_v1_caches
 
+User = get_user_model()
+
 
 @shared_task
 def update_api_caches():
     update_api_v1_caches()
     update_api_experimental_caches()
+
+
+def _create_ghost_user_username(id_: str) -> str:
+    return f"Ghost User {id_}"
+
+
+def _create_ghost_user_email(id_: str) -> str:
+    # The ID used in ghost user usernames and emails are not the same as their `User.id`
+    # This is because `User` still uses an autoincrement ID
+    return f"{id_}.gu@thunderstore.io"
+
+
+@shared_task
+def delete_user(user_id: int) -> None:
+    """
+    Delete a User.
+
+    This should be the only place where User.delete() is called.
+    """
+    with transaction.atomic():
+        user = User.objects.get(id=user_id)
+        if not user.comments.exists():
+            return
+        uuid = generate_ulid_as_uuid()
+        username = _create_ghost_user_username(uuid.hex)
+        email = _create_ghost_user_email(uuid.hex)
+        ghost_user = User.objects.create_user(username, email=email)
+        for comment in user.comments.iterator():
+            comment.author = ghost_user
+            comment.save(update_fields=("author",))
+        user.delete()
```

Comment:
This could be moved to become recursive so e.g. 2000 comments are deleted at once then the task calls itself. This might be overkill though

Reply:
Shouldn't be needed, it's fine to have long running background tasks. We can optimize task queue configurations if it becomes an issue, no need to worry about it in the application code.